### ISSUE TITLE
Bogus capitalization warning involving SAFE_CAST, NULL, and AS

### ISSUE THREAD

**Issue by barrywhart** [alanmcruickshank/sqlfluff#200]
Input #1 :
```
SELECT
    SAFE_CAST(NULL AS STRING) AS age_label
FROM
    `table`
```

Input #2:
```
SELECT
    NULL AS age_label
FROM
    `table`
```

Either of these files results in a similar warning:
```
$ sqlfluff lint --exclude-rules L001 --dialect bigquery test2.sql
== [test2.sql] FAIL
L:   2 | P:  34 | L014 | Inconsistent capitalisation of unquoted identifiers.
```

I'm not sure if the `SAFE_CAST` is important or not, but am including it in case it's a different parsing pattern, etc.

@alanmcruickshank: I'd be happy to work on some tickets if you'd like to suggest any (this one or others)! February was busy for me, but I'm starting to have some discretionary time again.

**Comment by alanmcruickshank** [comment]
Very strange! Thanks for raising. 

What it's flagging up is that it feels like `age_label` is inconsistently capitalised. However in both of your examples there aren't any other identifiers for it to be inconsistent with!

Do you have a config file at all or are you just running with the stock config (apart from the command line arguments)?

**Comment by barrywhart** [comment]
I have a configuration file -- I'll post it later.

**Comment by barrywhart** [comment]
This is my `.sqlfluff` file:
```
[sqlfluff]
verbose = 0
nocolor = False
dialect = bigquery
templater = jinja
rules = None
exclude_rules = None
recurse = 0

[sqlfluff:templater:jinja:macros]
# Macros provided as builtins for dbt projects
dbt_ref = {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
dbt_config = {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}

# Some rules can be configured directly from the config common to other rules.
[sqlfluff:rules]
tab_space_size = 4

# Some rules have their own specific config.
[sqlfluff:rules:L010]
capitalisation_policy = consistent

[sqlfluff:rules:L016]
max_line_length = 100

[sqlfluff:templater:jinja:context]
```

**Comment by alanmcruickshank** [comment]
Yep, silly error. The `NULL` in your query was being parsed as an identifier. It is capitalised and so when we reach `age_label`, it is flagged as an issue because it is lowercase. A PR is coming so that `NULL` is correctly parsed as a _keyword_ rather than an _identifier_.
